elementary/monitor/data_monitoring/alerts/integrations/teams/teams.py

```diff
@@ -0,0 +1,632 @@
+    RESULT_MESSAGE_FIELD,
+    TEST_PARAMS_FIELD,
+    TEST_QUERY_FIELD,
+    TEST_RESULTS_SAMPLE_FIELD,
+]
+
+STATUS_DISPLAYS: Dict[str, Dict] = {
+    "fail": {"display_name": "Failure"},
+    "warn": {"display_name": "Warning"},
+    "error": {"display_name": "Error"},
+}
+
+
+class TeamsIntegration(BaseIntegration):
+    def __init__(
+        self,
+        config: Config,
+        tracking: Optional[Tracking] = None,
+        override_config_defaults=False,
+        *args,
+        **kwargs,
+    ) -> None:
+        self.config = config
+        self.tracking = tracking
+        self.override_config_defaults = override_config_defaults
+        super().__init__()
+
+        # Enforce typing
+        self.client: TeamsClient
+
+    def _initial_client(self, *args, **kwargs) -> TeamsClient:
+        teams_client = TeamsClient.create_client(
+            config=self.config, tracking=self.tracking
+        )
+        if not teams_client:
+            raise Exception("Could not create a Teams client")
+        return teams_client
+
+    @staticmethod
+    def _get_alert_sub_title(
+        alert: Union[
+            TestAlertModel,
+            ModelAlertModel,
+            SourceFreshnessAlertModel,
+        ],
+    ) -> str:
+        subtitle = "**"
+        subtitle += f"Status: {alert.status}"
+        if alert.suppression_interval:
+            subtitle += f"   |   Time: {alert.detected_at_str}"
+            subtitle += (
+                f"   |   Suppression interval: {alert.suppression_interval} hours"
+            )
+        else:
+            subtitle += f"   |   {alert.detected_at_str}"
+        subtitle += "**"
+
+        return subtitle
+
+    @staticmethod
+    def _get_potential_action(reportlink: ReportLinkData):
+        action = potentialaction(reportlink.text)
+        action.addOpenURI(
+            reportlink.text,
+            [{"os": "default", "uri": reportlink.url}],
+        )
+        return action
+
+    def _add_report_link_if_applicable(
+        self,
+        alert: Union[
+            TestAlertModel,
+            ModelAlertModel,
+            SourceFreshnessAlertModel,
+            GroupedByTableAlerts,
+        ],
+    ):
+
+        if isinstance(alert, ModelAlertModel) or isinstance(
+            alert, SourceFreshnessAlertModel
+        ):
+            reportlink = get_model_runs_link(alert.report_url, alert.model_unique_id)
+        elif isinstance(alert, TestAlertModel):
+            reportlink = get_test_runs_link(
+                alert.report_url, alert.elementary_unique_id
+            )
+        elif isinstance(alert, GroupedByTableAlerts):
+            if not alert.model_errors:
+                reportlink = get_model_test_runs_link(
+                    alert.report_url, alert.model_unique_id
+                )
+            else:
+                reportlink = None
+        else:
+            reportlink = None
+
+        if reportlink:
+            action = self._get_potential_action(reportlink)
```

Comment:
Just remembered I recently added this method to AlertModel for that exact same reason :) 
So it can be just-
```suggestion
        report_link = alert.get_report_link()

        if report_link:
            action = self._get_potential_action(report_link)
```

Reply:
Implemented! Much cleaner indeed